src/storage/rdb_ziplist.cc

```diff
@@ -152,3 +148,63 @@ Status ZipList::peekOK(size_t n) {
 }
 
 uint32_t ZipList::getEncodedLengthSize(uint32_t len) { return len < ZipListBigLen ? 1 : 5; }
+
+uint32_t ZipList::ZipStorePrevEntryLengthLarge(unsigned char *p, unsigned int len) {
+  uint32_t u32 = 0;
+  if (p != nullptr) {
+    p[0] = ZipListBigLen;
+    u32 = len;
+    memcpy(p + 1, &u32, sizeof(u32));
+    memrev32ifbe(p + 1);
```

Comment:
Just curious why memrev after memcpy? Wouldn't that be faster to rev the register before writing into memory?

Reply:
Currently, it seems that Redis is implemented in the same way. I suspect it is to avoid the risks associated with operating directly on the original data.